### PR TITLE
Recurring expense UI changes 

### LINKED ISSUE #143: Budget does not update on UI when setting from 0 to any other value upon startup.
As mentioned in the title.

Bug can be replicated by the following sequence of steps:

(If fresh installation):
1. Simply start up the app.
2. Set a budget. 

(If you have pre-existing data):
1. Set the budget to 0.
2. Close the app and reopen the app.
3. Set the budget to anything other than 0.

### PR DESCRIPTION
**Pull request by nicleejy**
Changes:
1. Added new recurring expense screen to display recurring expenses.
2. Updated each command result to receive an enum of ScreenType instead of a boolean to facilitate the screen switching behavior between the commands.
3. Linked result details filters UI to the currently applied filters (category and timespan)
4. Updated the budget advice to show a warning text to the user if the user has exceeded their monthly budget

Bug fixes:
Resolve budget not incrementing from 0 after setting to a non-zero value on startup of application

closes #143 #96 #113

### CODE REVIEW

File: src/main/java/seedu/address/ui/ScreenType.java
Comment (jinbesan) [inline src/main/java/seedu/address/ui/ScreenType.java:1]: Well-needed refactoring of code!

**Review by jinbesan — COMMENTED** [review]
Looks good

**Comment by shirsho-12** [comment]
closes #129 closes #98 closes #96 closes #94 closes #93

File: src/main/java/seedu/address/logic/commands/ListCommand.java [hunk 1]
```diff
@@ -44,20 +48,27 @@ public ListCommand(Optional<ExpenseInCategoryPredicate> categoryPredicate,
...
-        if (categoryPredicate.isPresent() && timespanPredicate.isPresent()) {
-            model.updateFilteredExpensesList(categoryPredicate.get().and(timespanPredicate.get()));
-        } else if (categoryPredicate.isPresent()) {
+        categoryPredicate.ifPresentOrElse(filter -> model.updateCategoryFilter(
+                filter.getCategory()), () -> model.updateCategoryFilter(null));
+        timespanPredicate.ifPresentOrElse(filter -> model.updateTimeSpanFilter(
+                filter.getTimespan()), () -> model.updateTimeSpanFilter(ParserUtil.Timespan.ALL));
+        Predicate<Expense> combinedPredicate = PREDICATE_SHOW_ALL_EXPENSES;
+        if (categoryPredicate.isPresent()) {
+            combinedPredicate = combinedPredicate.and(categoryPredicate.get());
```
Comment (shirsho-12) [inline src/main/java/seedu/address/logic/commands/ListCommand.java:57]: good use of streams

File: src/main/java/seedu/address/model/AnalyticModelManager.java [hunk 2]
```diff
@@ -22,21 +19,20 @@
...
+    private final DoubleProperty monthlyRemaining;
+    private final DoubleProperty weeklySpent;
+    private final DoubleProperty weeklyRemaining;
+    private final DoubleProperty monthlyChange;
+    private final DoubleProperty weeklyChange;
+    private final DoubleProperty totalSpent;
+    private final DoubleProperty budgetPercentage;
+    private final DoubleProperty monthlyBudget;
+    private final DoubleProperty weeklyBudget;
+    private final LocalDate currentDate;
```
Comment (shirsho-12) [inline src/main/java/seedu/address/model/AnalyticModelManager.java:36]: may need to be refactored into separate class

**Pull request merged**